tests/scripts/thread-cert/border_router/MATN_05_ReregistrationToSameMulticastGroup.py

```diff
@@ -0,0 +1,244 @@
+from pktverify import packet_verifier, packet_filter
+from pktverify.consts import MA1
+import config
+import thread_cert
+
+# Test description:
+# The purpose of this test case is to verify that a Primary BBR (DUT) can manage
+# a re-registration of a device on its network to remain receiving multicasts.
+# The test also verifies the usage of UDP multicast packets across backbone and
+# internal Thread network.
+#
+# Topology:
+#    ----------------(eth)------------------
+#           |                  |     |
+#          BR_1 (Leader) ---- BR_2  HOST
+#           |                  |
+#           |                  |
+#          Router_1 -----------+
+#
+
+BR_1 = 1
+BR_2 = 2
+ROUTER_1 = 3
+HOST = 4
+
+MLR_TIMEOUT_MIN = 300
+
+
+class MATN_05_ReregistrationToSameMulticastGroup(thread_cert.TestCase):
+    USE_MESSAGE_FACTORY = False
+
+    TOPOLOGY = {
+        BR_1: {
+            'name': 'BR_1',
+            'is_otbr': True,
+            'allowlist': [BR_2, ROUTER_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        BR_2: {
+            'name': 'BR_2',
+            'allowlist': [BR_1, ROUTER_1],
+            'is_otbr': True,
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        ROUTER_1: {
+            'name': 'Router_1',
+            'allowlist': [BR_1, BR_2],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        HOST: {
+            'name': 'Host',
+            'is_host': True
+        },
+    }
+
+    def test(self):
+        br1 = self.nodes[BR_1]
+        br2 = self.nodes[BR_2]
+        router1 = self.nodes[ROUTER_1]
+        host = self.nodes[HOST]
+
+        br1.set_backbone_router(reg_delay=10, mlr_timeout=MLR_TIMEOUT_MIN)
+        br1.start()
+        self.simulator.go(5)
+        self.assertEqual('leader', br1.get_state())
+        self.assertTrue(br1.is_primary_backbone_router)
+
+        router1.start()
+        self.simulator.go(5)
+        self.assertEqual('router', router1.get_state())
+
+        br2.start()
+        self.simulator.go(5)
+        self.assertEqual('router', br2.get_state())
+        self.assertFalse(br2.is_primary_backbone_router)
+
+        host.start(start_radvd=False)
+        self.simulator.go(10)
+
+        # Router_1 registers for multicast address, MA1, at BR_1.
+        router1.add_ipmaddr(MA1)
+        self.simulator.go(5)
+
+        # 1. Host sends a ping packet to the multicast address, MA1.
+        self.assertTrue(
+            host.ping(MA1, backbone=True, ttl=10, interface=host.get_ip6_address(config.ADDRESS_TYPE.ONLINK_ULA)[0]))
+
+        # 4. Within MLR_TIMEOUT_MIN seconds, Host sends a ping packet to the
+        # multicast address, MA1. The destination port 5683 is used for the UDP
+        # Multicast packet transmission.
+        self.assertTrue(
+            host.ping(MA1, backbone=True, ttl=10, interface=host.get_ip6_address(config.ADDRESS_TYPE.ONLINK_ULA)[0]))
```

Comment:
Step 4 should use a UDP ping? 
UDP ping means send a UDP message to the target IP. 
See also: https://github.com/openthread/openthread/blob/main/src/cli/README_UDP.md